### PR TITLE
Slider - adding 'migrate' prop to allow using Incubator.Silder

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
Slider, GradientSlider, ColorSliderGroup - adding 'migrate' prop to allow using Incubator Silder instead of the old implementation.
Included: a fix for 'thumbTintColor' in the Incubator Slider to match old implementation.

## Changelog
Slider, GradientSlider, ColorSliderGroup - adding 'migrate' prop to allow using Incubator Silder instead of the old implementation. NO breaking changes.

### CODE REVIEW

File: src/components/slider/index.tsx [hunk 1]
```diff
@@ -120,6 +121,10 @@ export type SliderProps = Omit<ThumbProps, 'ref'> & {
    * The slider's test identifier
    */
   testID?: string;
+  /** 
+   * Whether to use the new Slider implementation using Reanimated
+   */
+  migrate?: boolean;
```
Comment (M-i-k-e-l) [inline src/components/slider/index.tsx:127]: We should add this to the API file

Comment (Inbal-Tish) [reply src/components/slider/index.tsx:127]: Done

File: src/incubator/Slider/index.tsx [hunk 2]
```diff
@@ -143,16 +149,18 @@ const Slider = React.memo((props: Props) => {
...
-    if (useRange) {
-      const maxValue = getValueForOffset(rangeThumbOffset.value,
-        trackSize.value.width,
-        minimumValue,
-        maximumValue,
-        stepXValue.value);
-      runOnJS(onRangeChangeThrottled)(value, maxValue);
-    } else {
-      runOnJS(onValueChangeThrottled)(value);
+    if (offset !== _prevOffset) {
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/incubator/Slider/index.tsx:152]: Yes, removed. The bug was on ColorSliderGroup which kept on updating even when the offset didn't change

File: src/components/slider/context/SliderGroup.tsx [hunk 3]
```diff
@@ -41,7 +41,9 @@ export default class SliderGroup extends Component<SliderGroupProps, SliderGroup
   render() {
     return (
       <View {...this.props}>
-        <SliderContext.Provider value={this.getContextProviderValue()}>{this.props.children}</SliderContext.Provider>
+        <SliderContext.Provider value={this.getContextProviderValue()}>
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/slider/context/SliderGroup.tsx:44]: I did it on purpose, I think it's clearer this way

File: src/components/slider/GradientSlider.tsx [hunk 4]
```diff
@@ -216,9 +217,12 @@ class GradientSlider extends Component<Props, GradientSliderState> {
         break;
     }
 
+    const SliderComponent = this.props.migrate ? NewSlider : Slider;
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/slider/GradientSlider.tsx:220]: Done

File: demo/src/screens/componentScreens/SliderScreen.tsx
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply demo/src/screens/componentScreens/SliderScreen.tsx:1]: I should not have pushed the changes on the old SliderScreen...  want to keep one with the old implementation and one with the new one for compression. I made the changes for testing and pushed it with the other files. I reverted these changes.

File: src/components/slider/context/SliderGroup.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/slider/context/SliderGroup.tsx:44]: I think it's against the rules 📏  but I'll leave it up to you

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**